privval/test_harness.go

```diff
@@ -0,0 +1,263 @@
+package privval
```

Comment:
It's pretty awesome to see this working :-) 
I think it is a bit unfortunate that we need to add a bunch of exported types and vars to the privval package, though. IMHO, the privval package ideally would not expose anything that isn't useful solely for the test-harness. For instance an internal package for this TestHarness would be cool but then we'd need to move it to the `command` package you've created. See https://golang.org/doc/go1.4#internalpackages

Reply:
Awesome, thanks!